docs/file-stores.md

```diff
@@ -0,0 +1,252 @@
+
+This document divides into these parts.
+
+- [Conceptual Overview](#conceptual-overview)
+- Using a file store from actions via a [Nimbella SDK](#using-file-stores-via-the-nimbella-sdk).
+   - [The `nodejs` SDK](#the-nodejs-sdk)
+   - [The `php` SDK](#the-php-sdk)
+   - [The `python` SDK](#the-python-sdk)
+- Managing a file store with the [`nim` CLI](#managing-file-stores-with-the-nim-CLI).
+
+### Conceptual overview
+
+An file store behaves like a file _system_ in many ways.
+
+- the entities in a file store are often called "files", and they have names, metadata, and contents, just like files in a filesystem
+- the names are hierarchical, separated by slashes and can be used to organize information into a hierarchy
+    - however, unlike in a file system, the "directories" are just portions names: they cannot be empty and are not created or deleted explicitly
+
+The two file stores provided with your Nimbella namespace are similar in some ways and specialized in others.
+
+- Only the web store provides permanent public URLs to its files (as opposed to time-limited signed URLs).  It is intended to hold the front end of your application.
+- The web store is directly provisioned from a Nimbella project when you issue `nim project deploy`.  There is no direct provisioning of the data store from a project.
+- Both can be accessed by using one of the Nimbella SDKs inside an action
+- Both can be queried and updated using `nim` 
+- Using the SDK, you can generate a _signed URL_ for a file in either store, allowing your application's front-end (or your end-users) a time-limited means to upload or download the contents of that file
+    - the upload capability can be useful in either store
+    - the download capability is useful for the data store: files in the web store already have world-accessible URLs
+
+### Using a file store via the Nimbella SDK
+
+Nimbella provides SDKs for use within actions.  Currently, these exist for `nodejs`, `php`, and `python`.   An SDK is under development for `java` and the plan is to eventually cover all supported languages.
+
+#### Using the `nodejs` SDK
+
+In an action written in JavaScript, you should include
+
+```
+const nim = require('@nimbella/sdk')
+```
+
+The `nim` object serves as a client factory for both file stores and for the key-value store.  Only the file store usage is covered here.  
+
+To obtain a client for use with your data store, use
+
+```
+const client = await nim.storage()  // client will access the data store
+// or equivalently
+const client = await nim.storage(false) // client will access the data store
+```
+
+You can also obtain a client for use with your web store.
+
+```
+const client = await nim.storage(true) // client will access the web store
+```
+
+The boolean argument is used to select which file store the client will work with.
+
+Note the `await` keyword in the examples.   The `storage` function returns a `Promise` resolving to a client, not the client itself.
```

Comment:
Nuance, this means the statement must appear inside an `async` function and not a top level.

Reply:
Will reword a bit accordingly.